relayer/chain/ethereum/syncer/cache.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2020 Snowfork
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package syncer
+
+import (
+	gethCommon "github.com/ethereum/go-ethereum/common"
+	gethTypes "github.com/ethereum/go-ethereum/core/types"
+)
+
+type HeaderCacheItem struct {
+	Header    *gethTypes.Header
+	Forwarded bool
+}
+
+// This is used to store the latest headers as they are published. Up to
+// `numHeightsToTrack` heights are stored. Once this number is reached, an old
+// height is pruned each time a new height is added. The current stored height
+// range is given by [minHeight, maxHeight].
+type HeaderCache struct {
+	headers           map[string]*HeaderCacheItem
+	hashesByHeight    map[uint64][]string
```

Comment:
To save the indirection of hashes, you could just store the pointers here too, i. e. 

```suggestion
	hashesByHeight    map[uint64][]*HeaderChacheItem
```